nixos/modules/services/networking/syncplay.nix

```diff
@@ -7,18 +7,37 @@ let
 
   cmdArgs =
     [ "--port" cfg.port ]
+    ++ optionals (cfg.isolateRooms) [ "--isolate-rooms" ]
+    ++ optionals (!cfg.ready) [ "--disable-ready" ]
+    ++ optionals (!cfg.chat) [ "--disable-chat" ]
     ++ optionals (cfg.salt != null) [ "--salt" cfg.salt ]
+    ++ optionals (cfg.motd != null) [ "--motd-file" "${pkgs.writeText "motd" cfg.motd}" ]
+    ++ optionals (cfg.motdFile != null) [ "--motd-file" cfg.motdFile ]
```

Comment:
I'm not entirely sure I understand what you mean with this, the two options allow for either using a string, or a file path, so they serve slightly different functions.

Reply:
What I mean is that you can implement `cfg.motd` with e.g. `services.syncplay.motdFile = lib.mkIf (cfg.motd != null) (builtins.toFile "syncplay-motd" cfg.motd);` rather than handling them both separately. It’s pretty common to do that kind of thing in NixOS modules, although sadly my memory and ripgrep have both failed me at turning up many explicit examples :) (But there’s e.g. `nixos/modules/services/monitoring/longview.nix`.)

(It could also be done locally, with e.g. `motdFile = if cfg.motdFile != null then cfg.motdFile else …`, but that seems less elegant to me. It’s also possible that other people with more experience with these kinds of settings than me would have more refined opinions about what the best way is, but it seems good to me to at least avoid duplicating the handling of the command‐line option, and to raise an error if both options are set, which I believe the approach I suggested should achieve.)

Reply:
I think I understand now, thanks.
I had to use `if then else` instead of `lib.mkIf` since the latter resolves to a set instead of a `string | null`, but I've set both of the `*File` instances to default to their raw option if set.